Lib/ntpath.py

```diff
@@ -135,7 +135,7 @@ def join(path, *paths):
             result_path = result_path + p_path
         ## add separator between UNC and non-absolute path
         if (result_path and not result_root and
-            result_drive and result_drive[-1:] not in colon + seps):
+            result_drive and result_drive[-1] not in colon + seps):
```

Comment:
Not sure this matters for performance a lot, but since `colon` is used once you could replace the definition of `colon` above with `colon_seps = ':' + seps` (similar for the bytes version) and write
```suggestion
            result_drive and result_drive[-1] not in colon_seps):
```
(but leaving as is, is also fine with me)